src/components/magnet/magnet.tsx

```diff
@@ -20,29 +22,92 @@ interface IProps {
 interface IState {
   isDragging: boolean;
   dragData: PIXI.interaction.InteractionData | null;
+  rotation: number;
+  alpha: number;
+  alpha2: number;
+  yOffset: number;
+  yOffset2: number;
 }
 
 export default class Magnet extends React.Component<IProps, IState> {
   constructor(props: IProps) {
     super(props);
     this.state = {
       isDragging: false,
-      dragData: null
+      dragData: null,
+      rotation: 0,
+      alpha: 0,
+      alpha2: 0,
+      yOffset: 0,
+      yOffset2: 40
     };
   }
 
+  public componentDidMount() {
+    this.props.app.ticker.add(this.tick);
+  }
+
+  public componentWillUnmount() {
+    this.props.app.ticker.remove(this.tick);
+  }
+
+  public calculateRotation = (delta: any) => {
+    if (this.props.rotation !== this.state.rotation) {
+      if (this.props.rotation === 180) {
+        return Math.min(180, this.state.rotation + 10 * delta);
+      } else {
+        return Math.max(0, this.state.rotation - 10 * delta);
+      }
+    } else {
+      return this.state.rotation;
+    }
+  }
+
+  public tick = (delta: any) => {
+    const newRotation = this.calculateRotation(delta);
+    let newYOffset = this.state.yOffset + 1 * delta;
+    if (newYOffset > 80) {
+      newYOffset = 0;
+    }
+    let newYOffset2 = this.state.yOffset2 + 1 * delta;
```

Comment:
It looks to me like these offsets are always the same, unless I'm missing something?

Reply:
They have different starting values so they are actually different, but, yes, we should probably be able to calculate both the offsets and the alphas from a single value.